Bottom: 34a91daf2067d0234f049b58aa339bf11db05d6d
Top:    1cf97ca29b60d2f04809e7b6ce706886935254b2
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-29 18:16:43 +0100

Refresh of debug-patch

---

diff --git a/src/autoroute.c b/src/autoroute.c
index 72fb2aa..8845aeb 100644
--- a/src/autoroute.c
+++ b/src/autoroute.c
@@ -1476,7 +1476,6 @@ showbox (BoxType b, Dimension thickness, int group)
       ddraw->draw_line (ar_gc, b.X1, b.Y2, b.X2, b.Y2);
       ddraw->draw_line (ar_gc, b.X1, b.Y1, b.X1, b.Y2);
       ddraw->draw_line (ar_gc, b.X2, b.Y1, b.X2, b.Y2);
-      ddraw->flush_debug_draw ();
     }
 
 #if 1
@@ -3688,6 +3687,8 @@ TracePath (routedata_t * rd, routebox_t * path, const routebox_t * target,
   RD_DrawLine (rd, -1, 0, 0, 0, 0, 0, NULL, false, false);
   if (TEST_FLAG (LIVEROUTEFLAG, PCB))
     gui->use_mask (HID_FLUSH_DRAW_Q);
+  if (ddraw != NULL)
+    ddraw->flush_debug_draw ();
 }
 
 /* create a fake "edge" used to defer via site searching. */
